drivers/vmwarevsphere/vsphere.go

```diff
@@ -473,8 +479,37 @@ func (d *Driver) Remove() error {
 		return err
 	}
 	if machineState == state.Running {
-		if err = d.Kill(); err != nil {
-			return fmt.Errorf("can't stop VM: %s", err)
+		if d.GracefulShutdownTimeout > 0 {
+			// graceful shutdown the machine
+			log.Infof("Gracefully shutting down VM %s (ID: %s)", d.MachineName, d.MachineId)
+			if err := d.Stop(); err != nil {
+				return err
+			}
+			// wait for machine to stop
+			elapsed := 0
+			for elapsed <= d.GracefulShutdownTimeout {
+				if machineState, err = d.GetState(); err != nil {
+					return err
+				}
+				if machineState == state.Stopped {
+					break
+				}
+				elapsed += gracefulShutdownSleep
+				time.Sleep(gracefulShutdownSleep * time.Second)
+				log.Debugf("Waiting for VM %s (ID: %s) to stop. Elapsed time: %d second(s)", d.MachineName, d.MachineId, elapsed)
+			}
```

Comment:
Sorry, another nit 😅 
```suggestion
			for elapsed := 0; elapsed <= d.GracefulShutdownTimeout; elapsed += gracefulShutdownSleep {
				if machineState, err = d.GetState(); err != nil {
					return err
				}
				if machineState == state.Stopped {
					break
				}

				time.Sleep(gracefulShutdownSleep * time.Second)
				log.Debugf("Waiting for VM %s (ID: %s) to stop. Elapsed time: %d second(s)", d.MachineName, d.MachineId, elapsed)
			}
```

Reply:
I thought about this before and did not do it this way because the value of `elapsed` is used outside the for loop.